README.md

```diff
@@ -24,3 +24,76 @@ fn main() -> Result<(), FormulaError> {
     Ok(())
 }
 ```
+
+## Formula Syntax Overview
+
+The formula engine supports simple arithmetic expressions that combine numbers, references to components, mathematical operators, and basic functions.
+
+### Numbers
+
+You can write integer or decimal numbers directly in formulas:
+
+```
+42
+3.14
+0.001
+```
+
+### Component References
+
+You can refer to microgrid components using **component references**. These are written as `#` followed by one or more digits:
+
+```
+#1     // Refers to component 1
+#42    // Refers to component 42
+```
+
+### Operators
+
+The following standard arithmetic operators are supported:
+
+- Addition: `+`
+- Subtraction: `-` (also supports unary minus, e.g., `-5`)
+- Multiplication: `*`
+- Division: `/`
+
+Expressions follow standard precedence rules (multiplication/division before addition/subtraction). Use parentheses to override precedence as needed:
+
+```
+#1 + 3 * #2         // Multiplies #2 by 3, then adds #1
+(#1 + 3) * #2       // Adds #1 and 3, then multiplies the result by #2
+```
+
+### Functions
+
+Formulas support these functions that operate on a comma-separated list of expressions:
+
+- `COALESCE(a, b, ...)` — Returns the first non-null value from the list
+- `MIN(a, b, ...)` — Returns the smallest value
+- `MAX(a, b, ...)` — Returns the largest value
+
+Examples:
+
+```
+COALESCE(#3, 0)         // Returns #3 if it's not null, otherwise 0
+MIN(#1, #2, 100)        // Returns the smallest of the three
+MAX(#1 + 2, #4 * 5)     // Evaluates both expressions, returns the larger
+```
+
+### Whitespace
+
+Whitespace is ignored and can be used freely to improve readability:
```

Comment:
Too bad we don't support adding comments in the formulas at the moment, that would be nice, especially for formulas that cover 50 components, etc., and we have to handwrite them for whatever reason.